CHANGELOG.md

```diff
@@ -44,6 +44,9 @@ Main (unreleased)
 
 - Allow converting labels to structured metadata with Loki's structured_metadata stage. (@gonzalesraul)
 
+### Other changes
+
+- Use Go 1.21.3 for builds. (@tpaschalis)
```

Comment:
I guess `agent-build-image:0.30.3` contains a new Go version? But how was `agent-build-image` updated? I'd expect it to be via this PR?

Reply:
cc @ptodev 

The [process](https://github.com/grafana/agent/tree/main/build-image) is 
a) Update the `build-image` directory with a PR like [this](https://github.com/grafana/agent/pull/5472)
b) Push a `build-image/v0.X.Y` tag
c) Wait for the automation to create and push the new build-image, make another PR to use it (like here).

Do you think it makes sense to document the process under docs/developer so people have some reference PRs to look at?

Reply:
Yes, thank you - it'd be great to have something in docs/developer about it.